### DISCUSSION TITLE
Search for similar Uuid with default ToSql?

### DISCUSSION THREAD

**Discussion by OscarCreator** [rusqlite/rusqlite#1472] [Questions] [answered]
I have a sqlite database

```sql
CREATE TABLE IF NOT EXISTS tasks (
  id TEXT PRIMARY KEY NOT NULL,
  name TEXT NOT NULL,
)
```

Where the id is a `Uuid`
```rust
struct Task {
  id: Uuid,
  name: String,
}
```

Is it possible to search for similar tasks with the default ToSql for Uuid?
```rust
conn.execute("
  SELECT * FROM tasks
  WHERE id LIKE ?1
  ", 
  params!["bee5d"]
)
```
I got it working if I saved the Uuid as a string `task.id.to_string()`, but I would like to use the default ToSql implementation for Uuid so I don't need to add extra parsing when reading from the database.

If I use the default ToSql for Uuid it can look like '(`pl>U' in the database.

**Comment by gwenn** [comment, marked as answer]
> Is it possible to search for similar tasks with the default ToSql for Uuid ?

I don't think so:
https://github.com/rusqlite/rusqlite/blob/30a211056e46920110070ca155d1794f0a8561ec/src/types/value.rs#L59

But you should be able to use the `task.id.as_bytes()[..6]` as parameter (not tested).
